internal/composer/openapi.go

```diff
@@ -0,0 +1,224 @@
+	UploadRequests []UploadRequest `json:"upload_requests"`
+}
+
+// ImageStatus defines model for ImageStatus.
+type ImageStatus struct {
+	Status         string          `json:"status"`
+	UploadStatuses *[]UploadStatus `json:"upload_statuses,omitempty"`
+}
+
+// Repository defines model for Repository.
+type Repository struct {
+	Baseurl string `json:"baseurl"`
+}
+
+// UploadRequest defines model for UploadRequest.
+type UploadRequest interface{}
+
+// UploadStatus defines model for UploadStatus.
+type UploadStatus interface{}
+
+// ComposeJSONBody defines parameters for Compose.
+type ComposeJSONBody ComposeRequest
+
+// ComposeRequestBody defines body for Compose for application/json ContentType.
+type ComposeJSONRequestBody ComposeJSONBody
+
+type ServerInterface interface {
+	// Create compose (POST /compose)
+	Compose(w http.ResponseWriter, r *http.Request)
+	// The status of a compose (GET /compose/{compose_id})
+	ComposeStatus(w http.ResponseWriter, r *http.Request)
+	// Get the OpenAPI JSON specification (GET /openapi.json)
+	GetOpenapiJson(w http.ResponseWriter, r *http.Request)
+}
+
+// Compose operation middleware
+func ComposeCtx(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+
+		next.ServeHTTP(w, r.WithContext(ctx))
+	})
+}
+
+// ComposeStatus operation middleware
+func ComposeStatusCtx(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+
+		var err error
+
+		// ------------- Path parameter "compose_id" -------------
+		var composeId string
+
+		err = runtime.BindStyledParameter("simple", false, "compose_id", chi.URLParam(r, "compose_id"), &composeId)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Invalid format for parameter compose_id: %s", err), http.StatusBadRequest)
+			return
+		}
+
+		ctx = context.WithValue(ctx, "composeId", composeId)
+
+		next.ServeHTTP(w, r.WithContext(ctx))
+	})
+}
+
+// GetOpenapiJson operation middleware
+func GetOpenapiJsonCtx(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+
+		next.ServeHTTP(w, r.WithContext(ctx))
+	})
+}
+
+// Handler creates http.Handler with routing matching OpenAPI spec.
+func Handler(si ServerInterface) http.Handler {
+	return HandlerFromMux(si, chi.NewRouter())
+}
+
+// HandlerFromMux creates http.Handler with routing matching OpenAPI spec based on the provided mux.
+func HandlerFromMux(si ServerInterface, r chi.Router) http.Handler {
+	r.Group(func(r chi.Router) {
+		r.Use(ComposeCtx)
+		r.Post("/compose", si.Compose)
+	})
+	r.Group(func(r chi.Router) {
+		r.Use(ComposeStatusCtx)
+		r.Get("/compose/{compose_id}", si.ComposeStatus)
+	})
+	r.Group(func(r chi.Router) {
+		r.Use(GetOpenapiJsonCtx)
+		r.Get("/openapi.json", si.GetOpenapiJson)
+	})
+
+	return r
+}
+
+// Base64 encoded, gzipped, json marshaled Swagger object
```

Comment:
Just... why? To save space in the source-code :question: Eh.. Not our decision, I guess.

Reply:
Yeah, no comment.